src/modules/host.rs

```diff
@@ -0,0 +1,40 @@
+use ansi_term::{Color, Style};
+use std::env;
+use std::process::Command;
+
+use super::{Context, Module};
+use std::ffi::OsString;
+
+/// Creates a module with the system hostname
+///
+/// Will display the hostname if all of the following criteria are met:
+///     - host.disabled is absent or false
+///     - host.ssh_only is false OR the user is currently connected as an SSH session (`$SSH_CONNECTION`)
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("host")?;
+
+    if module.config_value_bool("disabled").unwrap_or(false) {
+        return None;
+    }
+
+    let ssh_connection = env::var("SSH_CONNECTION").ok();
+    if module.config_value_bool("ssh_only").unwrap_or(true) && ssh_connection.is_none() {
+        return None;
+    }
+
+    let os_hostname: OsString = gethostname::gethostname();
+
+    let (host, style) = match os_hostname.into_string() {
+        Ok(host) => (host, Color::Green.bold().dimmed()),
+        Err(_) => (String::from("<invalid UTF>"), Color::Red.bold()),
```

Comment:
We generally don't display error conditions directly in the prompt. 

The preferred technique (that I've seen so far) is to make the module `None` in the event of an error, and then use `log::debug()` to emit a logging message that can be seen if the user runs with the right envars. See, for example, [battery.rs:23](https://github.com/starship/starship/blob/08aef016cdb1dbf53fc50b6e31ab0add2f40856b/src/modules/battery.rs#L23)